internal/pkg/build/remotebuilder/remotebuilder.go

```diff
@@ -147,7 +147,9 @@ func (rb *RemoteBuilder) Build(ctx context.Context) (err error) {
 
 		imageRef := library.NormalizeLibraryRef(bi.LibraryRef)
 
-		if err = library.DownloadImageNoProgress(ctx, c, rb.ImagePath, imageRef); err != nil {
+		// TODO - remotebuilder is *only* amd64 right now, but this will change
+		// in future.
+		if err = library.DownloadImageNoProgress(ctx, c, rb.ImagePath, "amd64", imageRef); err != nil {
```

Comment:
but then there should be a check locally that if a remote build is requested, then the requested architecture _must_ be amd64, no?

Reply:
there is no `--arch` flag for build, only pull. We can't request something else yet. Any remote build that is run (from any host) will end up as an amd64 image.

Reply:
After chatting with @EmmEff who is working on the builder in this area I will need to enable the `--arch` flag only in combination with the `--remote` option of `singularity build`, and will pass-through architecture requests in the `BuilderRequirements` map to future-proof things even if alternate archs aren't handled by the builder right now.

Reply:
Ok. Does this need changes in this PR or not? I just want to make sure nothing is falling thru the cracks.

Reply:
@mem - yes there will be changes to this PR - hopefully this afternoon, maybe tomorrow morning.